src/main/kotlin/org/rust/ide/navigation/goto/RsGotoSuperHandler.kt

```diff
@@ -49,7 +49,11 @@ fun gotoSuperTarget(source: PsiElement): NavigatablePsiElement? {
     }
 
     val mod = modOrMethod as RsMod
-    val superMod = mod.`super` ?: return null
-    return superMod.itemsAndMacros.filterIsInstance<RsModDeclItem>().find { it.reference.resolve() == mod }
-        ?: superMod
+    return when (mod) {
+        is RsFile -> mod.declaration
+        else -> {
+            val superMod = mod.`super` ?: return null
+            superMod.itemsAndMacros.filterIsInstance<RsModDeclItem>().find { it.reference.resolve() == mod } ?: superMod
```

Comment:
@matklad we can find `RsModDeclItem` only in `RsFile`, can't we?
If it's true, then it should be simplified to
```kotlin
return when (mod) {
    is RsFile -> mod.declaration
    else -> mod.`super`
}
```

Reply:
>we can find RsModDeclItem only in RsFile, can't we?

Nope :) 

```
fn foo() {
    #[path="bar.rs"]
    mod bar;
}
```

Reply:
My gut feeling is that it can be completely removed, because we actually find `modDecl` when we are evaluating `` `super` ``.

Reply:
Do you mean just return `mod.super` here?
Then we will navigate incorrectly in next case
```rust
//- foo/qqq.rs
/*caret*/

//- main.rs
mod foo { // <- we will go here
    #[path="qqq.rs"]
    mod bar; // <- but should go here
}
```

But looks like that
```rust
return when (mod) {
    is RsFile -> mod.declaration
    else -> mod.`super`
}
```
works as expected.
I've added some tests with `path` attribute

Reply:
Yeah, the current implementation looks great: we now longer do the work of finding the declaration twice, and we don't encode the same logic in several functions anymore!